File: 프로그래머스/2/131704. 택배상자/택배상자.py
def solution(order):
    s = []
    i = 0
    n = 0 
    while i < len(order) :
        if order[i] > n :
            n += 1
            s.append(n)
        elif s[-1] == order[i] :
            s.pop()
            i += 1
        else :
            return i
        
    return i

File: 프로그래머스/2/131704. 택배상자/택배상자.js
function solution(order) {
    var answer = 0;
    let i = 0;
    
    let stack = [];
    
    const len = order.length;
    
    while(answer < len) {
        if (order[answer] > i) {
            i += 1;
            stack.push(i);
        } else if (stack[stack.length - 1] === order[answer]){
            stack.pop();
            answer += 1;            
        } else {
            return answer;
        }
    }
    
    return answer;
}